labs/src/main/scala/org/scalalabs/basic/lab02/CollectionExercise.scala

```diff
@@ -32,8 +32,26 @@ object CollectionExercise01 {
    * Case 3: so it is okay if you want to just give up
    *
    */
+  
+  // After realizing the sample inputs/outputs provide enough information
+  // to figure out the mapping, I checked Prof Ben's solution to make sure
+  // there isn't actually any sort of consistent set of operations
+  // we can generally apply to each input character to get the correct output character.
+  // My intention before seeing his solution was to hardcode the mapping by
+  // indicating the mapping for each individual character,
+  // but I noticed that he uses the samples themselves to generate most of the mapping;
+  // this is my own attempt at that idea without referencing his code.
+  val in1 = "ejp mysljylc kd kxveddknmc re jsicpdrysi"
+  val out1 = "our language is impossible to understand"
+  val in2 = "rbcpc ypc rtcsra dkh wyfrepkym veddknkmkrkcd"
+  val out2 = "there are twenty six factorial possibilities"
+  val in3 = "de kr kd eoya kw aej tysr re ujdr lkgc jv"
+  val out3 = "so it is okay if you want to just give up"
+  
+  val translator: Map[Char, Char] = ("qz" ++ in1 ++ in2 ++ in3).zip("zq" ++ out1 ++ out2 ++ out3).toMap
```

Comment:
Both you and Prof. Ben use this same initial mapping value, but only one of those (`'q'→'z'`) is provided in the initial comments. The only way to know the reverse (`'z'→'q'`) at compile time is to construct the whole mapping anyway. The sample solution constructs the entire thing without that help by taking advantage of the map being a bijection, but this solution still requires all the work from building the mapping anyway and gains nothing by not compiling it in.